osu.Game.Rulesets.Catch/Mods/CatchModShrinking.cs

```diff
@@ -0,0 +1,97 @@
+using osu.Game.Rulesets.UI;
+using osuTK;
+
+namespace osu.Game.Rulesets.Catch.Mods
+{
+    public class CatchModShrinking : Mod, IUpdatableByPlayfield, IApplicableToBeatmap, IApplicableToTrack
+    {
+        public override string Name => "Shrinking Playfield";
+        public override string Acronym => "SP";
+        public override string Description => "The playfield will shrink over time.";
+        public override double ScoreMultiplier => 1;
+        public override IconUsage? Icon => FontAwesome.Solid.CompressArrowsAlt;
+
+        /// <summary>
+        /// The point in the beatmap at which the final ramping rate should be reached.
+        /// </summary>
+        public const double FINAL_RATE_PROGRESS = 0.75f;
+
+        private double finalRateTime;
+        private double beginRampTime;
+
+        [SettingSource("Initial scale", "The initial scale of the playfield.")]
+        public BindableNumber<double> InitialScale { get; } = new BindableDouble(1)
+        {
+            Default = 1,
+            Value = 1,
+            MinValue = 0.25,
+            MaxValue = 1,
+            Precision = 0.01,
+        };
+
+        [SettingSource("Final scale", "The playfield scale to ramp towards.")]
+        public BindableNumber<double> FinalScale { get; } = new BindableDouble(0.4)
+        {
+            Default = 0.4,
+            Value = 0.4,
+            MinValue = 0.2,
+            MaxValue = 0.95,
+            Precision = 0.01,
+        };
+
+        [SettingSource("Axis", "The axes to scale.")]
+        public Bindable<Axes> ActiveAxes { get; } = new Bindable<Axes>();
+
+        private ITrack track;
+
+        public void ApplyToTrack(ITrack track)
+        {
+            this.track = track;
+        }
+
+        public virtual void ApplyToBeatmap(IBeatmap beatmap)
+        {
+            double firstObjectStart = beatmap.HitObjects.FirstOrDefault()?.StartTime ?? 0;
+            double lastObjectEnd = beatmap.HitObjects.LastOrDefault()?.GetEndTime() ?? 0;
+
+            beginRampTime = firstObjectStart;
+            finalRateTime = firstObjectStart + FINAL_RATE_PROGRESS * (lastObjectEnd - firstObjectStart);
+        }
+
+        public void Update(Playfield playfield)
+        {
+            playfield.Anchor = Anchor.Centre;
+            playfield.Origin = Anchor.Centre;
+
+            double amount = (track.CurrentTime - beginRampTime) / Math.Max(1, finalRateTime - beginRampTime);
+            float scale = (float)(InitialScale.Value + (FinalScale.Value - InitialScale.Value) * Math.Clamp(amount, 0, 1));
+
+            switch (ActiveAxes.Value)
+            {
+                case Axes.Both: playfield.Scale = new Vector2(scale, scale); break;
+                case Axes.X: playfield.Scale = new Vector2(scale, 1); break;
+                case Axes.Y: playfield.Scale = new Vector2(1, scale); break;
+            }
+        }
+
+        public enum Axes
+        {
+            Both,
+            X,
+            Y
+        }
```

Comment:
Can definitely reuse the existing `osu.Framework.Graphics.Axes`, which is the case in other mods which use framework enums (like "barrel roll" using `RotationDirection`).

Reply:
Didn't know that existed but it seems to be exactly what I needed, but how do I remove the `None` option from the settings?

Reply:
Seems that isn't possible, at least define an enum that's less generic, something like:
```csharp
public enum ShrinkType
{
    Horizontal,
    Vertical,
    Both
}
```